### ISSUE TITLE
torch_randn_like function potentially giving different results with torch.manual_seed()?

### ISSUE THREAD

**Issue by OmegaAsh** [pytorch/pytorch#125491]
### 🐛 Describe the bug

Hi guys, I am not sure if I am missing something really obvious here. I am training a reinforcement learning algorithm, injecting exploration noise into the process. I found that I was getting different results each run of the algorithm, where I kept all seeds and parameters the same. However, when I specifically remove the noise calculation here, the problem is fixed.

```
torch.manual_seed(42)
noise = (torch.randn_like(action) * self.hp.target_policy_noise).clamp(-self.hp.noise_clip, self.hp.noise_clip)
next_action = (self.actor_target(next_state, fixed_target_zs) + noise).clamp(-1,1)
            
```

I am confused, since I am setting the seed, the size of the action variable is constant, and the self.hp.target_policy_noise and self.hp.noise_clip variables are also constant. I have also verified that setting the noise variable to a constant also fixes the problem.

If there does not seem to be anything wrong, I will put it down to something wrong with other parts of my code.

Thanks for your help,

### Versions

PyTorch version: 2.1.2+cu121

**Comment by OmegaAsh** [comment]
Hi guys, the issue is solved, and it seems to be an issue with the version 2.1.2, and updating to 2.3.0 fixed the issue. Thanks
